src/librustc_interface/passes.rs

```diff
@@ -838,7 +838,21 @@ fn analysis(tcx: TyCtxt<'_>, cnum: CrateNum) -> Result<()> {
 
     sess.time("MIR_effect_checking", || {
         for def_id in tcx.body_owners() {
-            mir::transform::check_unsafety::check_unsafety(tcx, def_id.to_def_id())
+            mir::transform::check_unsafety::check_unsafety(tcx, def_id.to_def_id());
+
+            // FIXME: make `requires_const_checking` a query?
```

Comment:
a query may be overkill, but a method on `TyCtxt` seems good. I think we need the same logic elsewhere?

Reply:
Yep. While doing the HIR const-checking. I'll refactor this.

Reply:
This is going to go in a separate PR that introduces a single `ConstContext` enum in the `hir` crate.